apps/cowswap-frontend/src/modules/appData/utils/typedHooks.ts

```diff
@@ -0,0 +1,53 @@
+import { buildAppDataHooks } from './buildAppDataHooks'
+
+import { AppDataHooks, CowHook, TypedAppDataHooks, TypedCowHook } from '../types'
+
+export function cowHookToTypedCowHook(hook: CowHook, type: TypedCowHook['type']): TypedCowHook {
+  return { ...hook, type }
+}
+
+export function typedAppDataHooksToAppDataHooks(hooks: TypedAppDataHooks | undefined): AppDataHooks | undefined {
+  if (!hooks) {
+    return hooks
+  }
+
+  const { pre, post, ...rest } = hooks
+
+  return {
+    ...rest,
+    pre: pre ? pre?.map(typedCowHookToCowHook) : undefined,
+    post: post ? post?.map(typedCowHookToCowHook) : undefined,
+  }
+}
+
+function typedCowHookToCowHook({ type: _, ...rest }: TypedCowHook): CowHook {
+  return rest
+}
+
+export function addPermitHookToHooks(hooks: TypedAppDataHooks | undefined, permit: CowHook): AppDataHooks | undefined {
```

Comment:
The "merger" or regular hooks with permit hooks.
Takes care of replacing whatever permit there was before, if any.